src/models/draft.ts

```diff
@@ -39,12 +40,22 @@ export default class Draft extends Message implements DraftProperties {
     this.initAttributes(props);
   }
 
+  fileIds(): (string | undefined)[] {
+    let fileIds: (string | undefined)[] = super.fileIds();
+    if (this.fileIdsToAttach) {
+      fileIds = fileIds
+        .concat(this.fileIdsToAttach)
+        .filter((item, idx, mergedArray) => mergedArray.indexOf(item) === idx);
```

Comment:
Why do we need this filter?

Reply:
To prevent duplication of file IDs. If a user saves a first version of a draft with two files of ids "file_1" and "file_2", then tries to update the draft and sets `this.fileIdsToAttach = ["file_2", "file_3"];` then without this filter the resulting array would be ["file_1", "file_2", "file_2", "file_3"] which I believe would result in an error on the API side.

Reply:
Gotcha, I think using a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) would be a bit better for de-duping items. 
eg.
```
const deduplicatedArr = Array.from(new Set(arrayWithDuplicates));
```

Reply:
@ozsivanov Beautiful! Done ✅